src/components/TodoList/TodoList.tsx

```diff
@@ -1,210 +1,129 @@
-                <span className="icon">
-                  <i className="far fa-eye" />
-                </span>
-              </button>
-            </td>
-          </tr>
-
-          <tr data-cy="todo" className="has-background-info-light">
-            <td className="is-vcentered">3</td>
-            <td className="is-vcentered"> </td>
-
-            <td className="is-vcentered is-expanded">
-              <p className="has-text-danger">fugiat veniam minus</p>
-            </td>
-
-            <td className="has-text-right is-vcentered">
-              <button data-cy="selectButton" className="button" type="button">
-                <span className="icon">
-                  <i className="far fa-eye-slash" />
-                </span>
-              </button>
-            </td>
-          </tr>
-
-          <tr data-cy="todo">
-            <td className="is-vcentered">4</td>
-            <td className="is-vcentered"><span className="icon" data-cy="iconCompleted"><i className="fas fa-check" /></span></td>
-            <td className="is-vcentered is-expanded">
-              <p className="has-text-success">et porro tempora</p>
-            </td>
-
-            <td className="has-text-right is-vcentered">
-              <button data-cy="selectButton" className="button" type="button">
-                <span className="icon">
-                  <i className="far fa-eye" />
-                </span>
-              </button>
-            </td>
-          </tr>
+  const dispatch = useDispatch();
+  const currentTodos = useAppSelector(state => state.currentTodo);
 
-          <tr data-cy="todo">
-            <td className="is-vcentered">5</td>
-            <td className="is-vcentered"> </td>
+  const isSelected = (todo: Todo) => currentTodos?.id === todo.id;
 
-            <td className="is-vcentered is-expanded">
-              <p className="has-text-danger">laboriosam mollitia et enim quasi adipisci quia provident illum</p>
-            </td>
+  const activeTodos = useAppSelector(({ filter, todos }) => {
+    if (filter.status === Filter.all && !filter.query) {
+      return todos;
+    }
 
-            <td className="has-text-right is-vcentered">
-              <button data-cy="selectButton" className="button" type="button">
-                <span className="icon">
-                  <i className="far fa-eye" />
-                </span>
-              </button>
-            </td>
-          </tr>
-
-          <tr data-cy="todo">
-            <td className="is-vcentered">6</td>
-            <td className="is-vcentered"> </td>
+    const lowQuery = filter.query.toLocaleLowerCase();
 
-            <td className="is-vcentered is-expanded">
-              <p className="has-text-danger">qui ullam ratione quibusdam voluptatem quia omnis</p>
-            </td>
+    return todos.filter(todo => {
+      if (filter.status === Filter.active && todo.completed) {
+        return false;
+      }
```

Comment:
```suggestion
    return todos.filter(todo =>
  (filter.status !== Filter.active || !todo.completed) &&
  (filter.status !== Filter.completed || todo.completed) &&
  todo.title.toLowerCase().includes(lowQuery)
);
```
you can shorten all this expression, also pls move it to eparate const and don't filter on return